analyzers/src/SonarAnalyzer.Common/Helpers/GeneratedCodeRecognizer.cs

```diff
@@ -99,13 +101,14 @@ private bool HasGeneratedCodeAttribute(SyntaxNode root)
 
         private static bool HasGeneratedFileName(SyntaxTree tree)
         {
-            if (string.IsNullOrEmpty(tree.FilePath))
-            {
-                return false;
-            }
-
             var fileName = Path.GetFileName(tree.FilePath).ToUpperInvariant();
             return GeneratedFileParts.Any(part => fileName.Contains(part));
         }
+
+        public static bool IsRazorGeneratedFile(SyntaxTree tree) =>
+            tree.FilePath.EndsWith("razor.g.cs", StringComparison.InvariantCultureIgnoreCase)
```

Comment:
Wouldn't it make more sense to use `StringComparison.OrdinalIgnoreCase`?

It just so happens that you don't have `ss` in the strings you are comparing against, as in `razor.g.ss`.

Otherwise, `razor.g.ß` would be detected as a razor-generated file.

Same in the lines below.